features/step_definitions/framework_application_steps.rb

```diff
@@ -112,6 +112,20 @@
   step "I visit the /suppliers/frameworks/#{framework_slug}/submissions/#{lot_slug}/previous-services page"
 end
 
+# Some suppliers have multiple services with the same name. Delete all existing
+# drafts with the same name as the service we want to add to avoid later
+# trouble finding links.
```

Comment:
Hmm, I can see why you're doing this, but I wonder if we could end up missing out on opportunities to test edge cases here? Do you think there are any situations where it might be important for us to have draft services with the same name in our test databases? I'm not sure, just spitballing here.

Reply:
I agree - this is not ideal.

The problem is that if there are multiple services with the same name, I've not found a way to work out which existing service each was copied from. Do you know if there is a way of doing this?

Reply:
Yeah I see the problem 🤔 . It just occurred to me though that on staging it won't be so much of an issue, because we refresh it weekly. So maybe just put a `TODO` note and leave it as-is for now? It's still an improvement.

Reply:
Yes. Better imperfect tests than none.

I'll create a story to improve the tests for the framework applications and include this problem. The other thing we will want to tackle is that we don't test lots that have a maximum of only one service (which is most lots for DOS)